### ISSUE TITLE
We should scan the results of a saved question and pull out information like FieldValues, etc

### ISSUE THREAD

**Issue by salsakran** [metabase/metabase#5594]
When a user saves a question, we should keep process the results of the query in the same way we analyze actual databases to do things like
* guess semantic type
* cache field values for filters
etc

**Comment by niparis** [comment]
Would that allow using filters (in dashboards) on questions built from saved questions (currently not possible)?

**Comment by ejhayes** [comment]
I agree, this functionality would be incredibly useful, especially for the data sandboxing functionality in the Enterprise version. Otherwise attempting to do filtering on a sandboxed question tends to result in a `Error: Table <id> is not present in the Query Processor Store` error.

**Comment by nickp** [comment]
Is there a workaround for this? It seems to be the case that you cannot filter columns due to this issue, unless you write a native SQL Question, which comes with its own costs (you also won't be able to filter the resulting table with all of the nice-to-haves that a table made with the GUI editor provides)

**Comment by JamesPoel** [comment]
This issue is very stale - do we have an update?

**Comment by bunny-rich** [comment]
+1 on this. We've had to switch from using DB views to SQL questions to get row level security/impersonation and now none of the drop-downs are populated with results.

**Comment by thebiglabasky** [comment]
Hi @bunny-rich (and others interested).
It seems like your looking specifically for ways to get filter drop downs available from SQL Questions, correct?
This issue is not specifically about that, but rather about incrementally improving fingerprinting data, column types detection, etc. (that includes refining field values we've already scanned) as queries get run, which wouldn't change the limitation on SQL questions.
The closest workaround to offer dropdown filters on SQL question today is to add [parameters](https://www.metabase.com/docs/latest/questions/native-editor/sql-parameters) on fields that you care about, and map these parameters to another question's result or a custom list. This isn't exactly meeting what I assume you'd like to have but can get you pretty close.
